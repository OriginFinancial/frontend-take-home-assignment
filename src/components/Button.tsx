import * as React from 'react';
import styled from 'styled-components';

const Button = styled.button`
  background: ${props => props.theme.buttons.default};
  border: 1px solid #1b31a8;
  border-radius: 32px;
  color: #fff;
  height: 56px;
  font-size: 18px;
  font-weight: 600;
  width: 100%;
`;

export default Button;
