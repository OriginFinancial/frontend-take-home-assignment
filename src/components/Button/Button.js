import styled, { css } from 'styled-components';
import { mq } from '../utils/mediaQueries';
import Sizes from '../utils/buttonSizes';

const buttonBase = ({ minWidth = 'md' }) => css`
  ${mq({ minWidth: ['40px', '80px', '100px', Sizes[minWidth]] })}
  padding: 12px 16px;
  color: #fff;
  cursor: pointer;
  border: none;
  border-radius: 25px;
  background-color: #1b31a8;
  font-size: 16px;
  font-weight: 600;
`;

const buttonSecondary = ({ rounded }) =>
  rounded &&
  css`
    ${mq({ minWidth: [0, 0, 0, 0, 0, 0] })}
    padding: 1em;
    color: #000;
    border-radius: 50%;
    background-color: #fff;
    border: 1px solid #e1e8ed;
  `;

const Button = styled.button`
  ${buttonBase}
  ${buttonSecondary}
`;

Button.displayName = 'Button';

export { Button };
