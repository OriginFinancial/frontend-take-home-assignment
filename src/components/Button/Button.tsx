import * as React from 'react';
import styled from 'styled-components';
import { ChildrenProps } from '../../models';

const Button: React.FunctionComponent<ChildrenProps> = ({ children }) => {
  return <Wrapper>{children}</Wrapper>;
};

const Wrapper = styled.button`
  background-color: ${props => props?.theme?.colors?.primary?.main};
  border: 0.1rem solid ${props => props?.theme?.colors?.primary?.main};
  border-radius: 3rem;
  color: ${props => props?.theme?.colors?.white};
  cursor: pointer;
  font-weight: 700;
  padding: 1.8rem 0.8rem;
  text-align: center;
  transition: background-color 0.4s;
  width: 100%;

  &:hover {
    background-color: ${props => props?.theme?.colors?.primary?.dark};
  }

  &:focus {
    outline: 0.1rem solid ${props => props?.theme?.colors?.primary?.light};
  }
`;

export default Button;
