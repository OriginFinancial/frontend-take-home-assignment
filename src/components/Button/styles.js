import styled from 'styled-components';

export const Button = styled.button`
  background-color: #1b31a8;
  color: white;
  height: 56px;
  border-radius: 32px;
  border: none;
  max-width: 400px;
  font-weight: 600;
`;
