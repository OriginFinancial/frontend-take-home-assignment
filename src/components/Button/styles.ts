import styled from 'styled-components';

export const StyledButton = styled.button`
  color: #fff;
  font-weight: bold;
  font-size: var(--text-md);
  letter-spacing: 0.0625rem;
  background: #1b31a8;
  padding: 1rem;
  min-width: 100%;
  display: block;
  margin: 0 auto;
  border: 0;
  border-radius: 2rem;
  outline: none;
  cursor: pointer;
  transition: 0.3s ease-in-out;

  &:hover,
  &:focus {
    background-color: #0179ff;
  }

  @media (min-width: 30em) {
    min-width: 25rem;
  }
`;
