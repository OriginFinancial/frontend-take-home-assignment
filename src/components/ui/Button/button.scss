@import '~/assets/styles/_colors.scss';
@import '~/assets/styles/_globalVars.scss';

.button {
  width: 80%;
  padding: 2vh 0;
  background: $color_primary;
  border-radius: 50px;
  margin: 0 auto;
}
