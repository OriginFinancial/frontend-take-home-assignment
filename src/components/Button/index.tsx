import React from 'react';
import { Button as SButton } from './styles';

const Button = props => <SButton {...props}>{props.children}</SButton>;

export default Button;
