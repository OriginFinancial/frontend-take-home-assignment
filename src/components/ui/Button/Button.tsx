import * as React from 'react';
import './button.scss';

interface Props {
  enable?: number;
}

const Button: React.FC<Props> = props => (
  <button className="Button" disabled={props.enable <= 0 ? true : false}>
    Confirm
  </button>
);

export default Button;
