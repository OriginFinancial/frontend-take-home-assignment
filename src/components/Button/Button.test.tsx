import * as React from 'react';
import { mount, ReactWrapper } from 'enzyme';
import { ChildrenProps } from '../../models';
import Button from './Button';

describe('Button component', () => {
  let component: ReactWrapper;
  let props: ChildrenProps;

  beforeEach(() => {
    jest.clearAllMocks();
    component = mount(<Button {...props}>Confirm</Button>);
  });

  it('renders initial layout', () => {
    expect(component.getElements()).toMatchSnapshot();
  });

  it('renders "Confirm" text passed as children', () => {
    expect(component.find('button').text()).toBe('Confirm');
  });
});
