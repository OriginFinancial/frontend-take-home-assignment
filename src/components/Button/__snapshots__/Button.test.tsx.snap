// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component renders initial layout 1`] = `
Array [
  <Button>
    Confirm
  </Button>,
]
`;
